### PR TITLE
[ch33586]: fix high memory usage phase II

### PR DESCRIPTION
**Pull request by hbarisik**

### CODE REVIEW

**Comment by domdinicola** [comment]
pls rebase this @hbarisik

**Comment by domdinicola** [comment]
@hbarisik fix conflicts pls

File: src/etools_datamart/apps/mart/data/models/audit_audit.py [hunk 1]
```diff
@@ -14,17 +16,26 @@
...
-            filters = self.config.key(self, record.engagement_ptr)
-            values = self.get_values(record.engagement_ptr)
-            op = self.process_record(filters, values)
-            self.increment_counter(op)
+        batch_size = settings.RESULTSET_BATCH_SIZE
+        logger.debug(f"Batch size:{batch_size}")
+
+        qs = AuditAudit.objects.select_related("engagement_ptr")
+
+        for batch_start in range(0, qs.count(), batch_size):
```
Comment (domdinicola) [inline src/etools_datamart/apps/mart/data/models/audit_audit.py:29]: can you check `iterator(self, chunk_size=None)`

File: CHANGES [hunk 2]
```diff
@@ -1,3 +1,7 @@
+4.4
+----
+* batch etl loading to reduce memory usage
+
```
Comment (domdinicola) [inline CHANGES:4]: pls resolve conflicts

File: src/etools_datamart/apps/mart/data/models/reports_indicator.py [hunk 3]
```diff
@@ -77,12 +82,18 @@ def get_pd_outputs(self, record, values, field_name, **kwargs):
...
         qs = self.filter_queryset(self.get_queryset())
-        for record in qs.all():
-            filters = self.config.key(self, record)
-            values = self.get_values(record)
-            op = self.process_record(filters, values)
-            self.increment_counter(op)
+
+        for batch_start in range(0, qs.count(), batch_size):
+            batch = qs[batch_start : batch_start + batch_size]
+            for record in batch.iterator():
```
Comment (domdinicola) [inline src/etools_datamart/apps/mart/data/models/reports_indicator.py:92]: can we run iterator with batch on the queryset and check performance?

qs.iterator(chunk_size=batch_size)

Comment (hbarisik) [reply src/etools_datamart/apps/mart/data/models/reports_indicator.py]: Based on the following work Paginator is a better choice. Django Paginatir has been used.
https://nextlinklabs.com/resources/insights/django-big-data-iteration

**Comment by domdinicola** [comment]
@hbarisik can I ask you to rebase and squash the commits?
did you test it locally?

**Comment by domdinicola** [comment]
@hbarisik pls confirm when this is working locally

**Pull request merged**